CONTRIBUTING.md

```diff
@@ -0,0 +1,70 @@
+
+We use [Github Issues](https://github.com/FairRootGroup/FairRoot/issues/new/choose).
+
+## Contributing code
+
+We use [Github Pull Requests](https://github.com/FairRootGroup/FairRoot/compare).
+
+Checklist:
+
+1. By default, develop against the [`dev`](https://github.com/FairRootGroup/FairRoot/tree/dev)
+   branch and base your PR against it.
+   * In rare cases (e.g. backports, some hotfixes) base against the appropriate
+     branch.
+2. Follow [the seven rules of a great Git commit message]
+   (https://cbea.ms/git-commit/#seven-rules)!
+   * Use a meaningful commit granularity, e.g. do not mix
+     unrelated changes in a single commit and vice verse squash related commits
+     into one.
+   * Expect that a reviewer will ask you for restructuring your commits! This
+     usually happens towards the end of the lifetime of a PR when it matured
+     enough for merging.
+   * Utilize [the commit message body to explain *what* and *why* and not *how*](https://cbea.ms/git-commit/#why-not-how).
+     Aim your message body content to be meaningful (concise, but complete
+     enough) to a reader in the future!
+   * Utilize [reference keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)
+     both in commit messages as well as in PR comments if applicable.
+3. If you are a first time contributor, add a separate commit in your PR which
+   adds your name to the [`CONTRIBUTORS`](CONTRIBUTORS) file.
+4. Follow our [Coding Guidelines](#coding-guidelines).
+5. Address failed CI checks.
+
+## Coding Guidelines
+
+These guidelines are horribly incomplete at this point in time, but one has to
+start somewhere ;) Exactly because these guidelines are relatively new, do not be
+alarmed if large parts of this repo do not follow them (yet). The idea is to
+apply them incrementally on new code and when existing code is modified. These
+guidelines will also evolve as time passes.
+
+### G - General
+
+1. Our CI enforces formatting with [`clang-format`](https://clang.llvm.org/docs/ClangFormat.html)
+   (see our [`.clang-format` config](.clang-format)). Rarely, it is needed to
+   opt-out of *clang-format* for certain code sections because it would just
+   make things worse.
+2. Our CI runs [`clang-tidy`](https://clang.llvm.org/extra/clang-tidy/) (see
+   our [`.clang-tidy` config](.clang-tidy)).
+3. Write readable, documented and maintainable code! (TODO be more precise)
+
+### N - Naming
+
+1. Prefer variable names describing what they contain, avoid repeating just the
+   typename.
+
+### C - Comments
+
+1. Use [doxygen-style comments](https://www.doxygen.nl/manual/commands.html)!
+
+### I - Initialization
+
+1. Prefer [default member initialization](https://en.cppreference.com/w/cpp/language/data_members#Member_initialization)
+   with braces, e.g.
+   ```cpp
+   class FairDetector {
+     int fDetId{0};
+   };
+   ```
```

Comment:
Maybe?

> Sometimes a classic member initializer is also fine. The obvious case is the specific constructor wanting to set a non default value. But also `unique_ptr`s of incomplete types (for what ever reason) should use member initializers and not default initializers for the time being.

Reply:
at least that is how I understood [your comment](https://github.com/FairRootGroup/FairRoot/pull/1273#discussion_r1023669941)

Reply:
Okay, I think, there are two points here:

1. Whether or not to recommend default member initializers at all, or not.
   * This one is by me. I like default member initializers, because they make constructors more concise, avoid member initializer order errors, and avoid repetitions for multiple constructors.
   * Not to mention C++ Core Guidelines [C.48](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#c48-prefer-in-class-initializers-to-member-initializers-in-constructors-for-constant-initializers)
   * But then there are some cases, where a "classic" member initializer is better. For one, if not using the _default_ value. For another, older compilers are not completely happy about `unique_ptr`s with an incomplete type.
2. If one actually uses a default member initializer, then there is the question of style. *I* didn't care much about it. When starting to use them, I used `{…};` for objects and ` = …;` for simple types. @fuhlig1 suggested to use one style for everything. Since using ` = …;` for complex objects feels very broken, the `{…};` style was left over.

Reply:
To me `= ...;` is more readable. Do we have to enforce consistency in this case?
I would vote for making `= ...;` the default/preferred one. And only use the other where it is unavoidable for some reason.

As I understand `= ...;` requires you to make things more explicit sometimes. To quote an example from @mattkretz:
```c++
class A
{
  public:
  explicit A(int);
};

class B
{
  A a1 = 1; // error
  A a2{1}; // ok
};
```
, which can still be solved by writing this instead:
```c++
class A
{
  public:
  explicit A(int);
};

class B
{
  A a1 = A(1);
  A a2 = A(1);
};
```

Reply:
Right, :+1: for using default member initializers (and then defaulting the default ctor :grin:).

But otherwise, don't restrict initialization style. All have their places. In C++ the most natural syntax for initialization, to me at least, appears to be `Type name = init;`. I've always found `Type name(init);` to be strange - and it leads to the most vexing parse, which is why we got `Type name{init};`. Note that the latter has different semantics, because initializer lists (i.e. braces) require non-narrowing conversions.

Reply:
I just removed this guideline from this PR. Note: It was also contended by two other persons during today's group meeting. So, it seems unlikely to be adopted at this point in time.